deployment/src/main/java/io/quarkiverse/quinoa/deployment/packagemanager/FrameworkType.java

```diff
@@ -112,6 +112,33 @@ public static DetectedFramework detectFramework(LaunchModeBuildItem launchMode,
         return new DetectedFramework(frameworkType, packageJson, startCommand);
     }
 
+    /**
+     * Check whether this path should be scanned for changes by comparing against known directories that should be ignored.
+     * Ignored directories include any that start with DOT "." like ".next" or ".svelte", also "node_modules" and any
+     * of the framework build directories.
+     *
+     * @param filePath the file path to check
+     * @return true if it is a directory that should be scanned for changes, false if it should be ignored
+     */
+    public static boolean shouldScanPath(Path filePath) {
+        if (!Files.isDirectory(filePath)) {
+            // not a directory so do not scan
+            return false;
+        }
+
+        final StringBuilder regex = new StringBuilder("[.]\\S+|node_modules|target");
```

Comment:
instead of appending the framework buildDir, why don't you just add a hard comparison?

That said, I would:
- Keep the static IGNORE_WATCH Set and merge the framework build dirs (static)
- Check if it should be ignored
- Use a static compiled REGEX with `^[.].+$`

Reply:
Changes made